base/doc/usrguide.tex

```diff
@@ -1258,4 +1258,39 @@ \section{Case changing}
   |\DeclareUppercaseMapping[xx]{"0131}{I}|
 \end{quote}
 
+\section{Support for problem solving}
+
+\begin{decl}
+  |\listfiles| \oarg{options}
+\end{decl}
+
+If this command is placed in the preamble then a list of the files
+read in (as a result of processing the document) will be displayed
+on the terminal (and in the log file) at the end of the run. Where
+possible, a short description will also be produced. These descriptions
+will (hopefully) include the descriptions, dates and version numbers
+for package and class files.
+
+Sometimes, if may be that a local edit has been made to a package or
+class file (or rather a copy of such a file). To allow these cases to
+be identified, \cs{listfiles} takes an optional argument which allows
+adjustment of the information printed using a key--value approach
+\begin{description}
+  \item[\texttt{hashes}] Adds the MD5 hash for each file to the
+    information printed
+  \item[\texttt{sizes}] Adds the file size for each file to the
+    information printed
+\end{description}
+Note that the difference in line ends used by Windows and Unix
+systems \emph{does} show up here: as such, you should compare
+results from computers using the same operating system.
```

Comment:
is that really an issue, I mean if somebody sends me such a listing from a Windows machine, would I have problems to compare it with my OSX output after copy-paste?

I mean if I get the full log then this will not be something one can compare across OSes anyway, because all the paths are likely to be different, but for something put into a github issue or send by mail ... ?

So I think the "show up here" needs some qualifying or perhaps the whole para is not needed because all of this was true also for the original \listfiles

Reply:
I've revised the text: the impact is that you have a different number of bytes and thus a different MD5 hash based on LF vs. LF CR.

Reply:
The alternative would be to read the files to a token list then hash that: TeX then deals with the system-dependence. But that will be slower ..

Reply:
ah, that is more understandable that the sizes and the hashes differ in that case, so "here" is in the hash and size result :-)

Reply:
To indicate the source of hash and size differences, maybe the current platform (`\c_sys_platform_str`) can be logged too, when one or more of "hashes" and "sizes" are specified.

Or even a full sentence,
```tex
 *File List*
    (Current platform is "unix". Note the sizes and hashed may differ between platforms due to different default line ends.)
 article.cls    2023/05/17 v1.4n Standard LaTeX document class
    (size: 20144, hash: FBCF51733730D8F1A62143512FF5B1FE)
  size10.clo    2023/05/17 v1.4n Standard LaTeX file (size option)
    (size: 8448, hash: 74078C4A887F2CDCD57C3EC111622740)
l3backend-luatex.def    2023-11-04 L3 backend support: PDF output (LuaTeX)
    (size: 30409, hash: 4F844A9024E5A9B01FC5871F268DC768)
  ts1cmr.fd    2023/04/13 v2.5m Standard LaTeX font definitions
    (size: 2430, hash: A2E8A746C2C6DDF1DF1FB6AD609D5F6A)
 ***********
```

Reply:
You have to watch it a bit: you _can_ have files with different line endings to the system default and they will work (as TeX normalises), so if someone has e.g. something by email then it will still work ...

Reply:
Oh, you're right.